core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -931,31 +932,57 @@ def _apply_conformalization(self, y_pred: np.ndarray) -> np.ndarray:
     def predict(self, X, **kwargs) -> np.ndarray:
         """
         Returns class predictions of X.
-        For binary and multiclass problems, this returns the predicted class labels as a Series.
-        For regression problems, this returns the predicted values as a Series.
+        For binary and multiclass problems, this returns the predicted class labels as a 1d numpy array.
+        For regression problems, this returns the predicted values as a 1d numpy array.
         """
         y_pred_proba = self.predict_proba(X, **kwargs)
         y_pred = get_pred_from_proba(y_pred_proba=y_pred_proba, problem_type=self.problem_type)
         return y_pred
 
-    def predict_proba(self, X, normalize=None, **kwargs) -> np.ndarray:
+    def predict_proba(self, X, *, normalize: bool | None = None, record_time: bool = False, **kwargs) -> np.ndarray:
```

Comment:
why do we not set the `record_time` as true and by default record the time taken for prediction?

Reply:
Because calling `time.time` takes time, and `predict` is something we want to take as little time as possible for the user.